src/Microsoft.TestPlatform.ObjectModel/Client/Events/StopTestSessionCompleteEventArgs.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+/// <summary>
+/// Event arguments used to notify the caller about the status of the test session.
+/// </summary>
+[DataContract]
+public class StopTestSessionCompleteEventArgs : EventArgs
+{
+    /// <summary>
+    /// Creates an instance of the current class.
+    /// </summary>
+    public StopTestSessionCompleteEventArgs()
+    { }
+
+    /// <summary>
+    /// Creates an instance of the current class.
+    /// </summary>
+    /// 
+    /// <param name="testSessionInfo">The test session info.</param>
+    public StopTestSessionCompleteEventArgs(TestSessionInfo testSessionInfo)
+    {
+        TestSessionInfo = testSessionInfo;
+    }
+
+    /// <summary>
+    /// Gets or sets the test session info.
+    /// </summary>
+    [DataMember]
+    public TestSessionInfo TestSessionInfo { get; set; } = null;
+
+    /// <summary>
+    /// Gets or sets the metrics.
+    /// </summary>
+    [DataMember]
+    public IDictionary<string, object> Metrics { get; set; } = null;
+
+    /// <summary>
+    /// Gets or sets a value indicating if the session was successfully stopped or not.
+    /// </summary>
+    [DataMember]
+    public bool IsStopped { get; set; } = false;
```

Comment:
Could we end up in a "unknow" state that would justify using a nullable bool here?

Reply:
We cannot end up in an unknown state, we either close the session successfully or we fail to close it because it's been already closed. I'd rather not introduce a nullable bool if we can avoid it as that would be an additional failure point that users of the `TranslationLayer` would have to guard against.